scipy/stats/_hypotests.py

```diff
@@ -395,51 +396,6 @@ def _get_wilcoxon_distr(n):
     return np.array(cnt, dtype=int)
 
 
-def _Aij(A, i, j):
-    """Sum of upper-left and lower right blocks of contingency table."""
-    # See [2] bottom of page 309
-    return A[:i, :j].sum() + A[i+1:, j+1:].sum()
-
-
-def _Dij(A, i, j):
-    """Sum of lower-left and upper-right blocks of contingency table."""
-    # See [2] bottom of page 309
-    return A[i+1:, :j].sum() + A[:i, j+1:].sum()
-
-
-def _P(A):
-    """Twice the number of concordant pairs, excluding ties."""
-    # See [2] bottom of page 309
-    m, n = A.shape
-    count = 0
-    for i in range(m):
-        for j in range(n):
-            count += A[i, j]*_Aij(A, i, j)
-    return count
-
-
-def _Q(A):
-    """Twice the number of discordant pairs, excluding ties."""
-    # See [2] bottom of page 309
-    m, n = A.shape
-    count = 0
-    for i in range(m):
-        for j in range(n):
-            count += A[i, j]*_Dij(A, i, j)
-    return count
-
-
-def _a_ij_Aij_Dij2(A):
-    """A term that appears in the ASE of Kendall's tau and Somers' D."""
-    # See [2] section 4: Modified ASEs to test the null hypothesis...
-    m, n = A.shape
-    count = 0
-    for i in range(m):
-        for j in range(n):
-            count += A[i, j]*(_Aij(A, i, j) - _Dij(A, i, j))**2
-    return count
```

Comment:
I think this cannot be removed yet, because Pythran is still optional. If `SCIPY_USE_PYTHRAN` is `0`, then there's a problem. I think this needs to be structured like:
```python
try:
    from _hypotests_pythran import _P, _Q, a_ij_Aij_Dij2
except ImportError:
    def _P(A):
        ...
    ...
```

Reply:
oh, I thought if `SCIPY_USE_PYTHRAN` is `0`, then the `_hypotests_pythran` file would not be compiled but stayed like a pure python file, and we can still import function from it?

Reply:
Argh, you are right and I should have more coffee! Of course:)

Reply:
sorry for the noise!